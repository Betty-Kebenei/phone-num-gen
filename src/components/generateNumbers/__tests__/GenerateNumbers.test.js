import React from 'react';
import renderer from 'react-test-renderer';
import {configure, mount } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';

import GenerateNumber from '../GenerateNumbers';

configure({ adapter: new Adapter() })

describe('GenerateNumber', () => {
  test('header component renders correctly', () => {
    const component = renderer.create(<GenerateNumber />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('handleSubmit', () => {
    test('quantity state gets updated', () => {
      const wrapper = mount(<GenerateNumber />);
      wrapper.setState({quantity: 0});
      wrapper.find('[name="quantity"]').simulate('change', {target: {name: 'quantity', value: 10}});
      wrapper.find('[type="submit"]').simulate('click');
      expect(wrapper.state().quantity).toEqual(10);
    })
  })

})

