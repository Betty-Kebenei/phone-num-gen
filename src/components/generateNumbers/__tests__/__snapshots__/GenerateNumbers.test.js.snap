// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenerateNumber header component renders correctly 1`] = `
<div
  className="generator"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        htmlFor="quantity"
      >
        How many numbers do you want to generate? 
      </label>
      <br />
      <input
        id="quantity"
        max={1000}
        min={10}
        name="quantity"
        onChange={[Function]}
        required={true}
        type="number"
        value={0}
      />
    </div>
    <input
      className="btn btn-primary"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
